docs2/site/docs/getting-started/custom-scalars.md

```diff
@@ -183,105 +213,303 @@ public class ExampleSchema : Schema
+autoincrementing numeric identifier internally, where null values are stored as 0.
 
-- `Serialize` is _also_ called when reading variables from the client so that variables can be converted to `IValue` instances. In the case of `Vector3Type`, `value` will be a string during this process.
+```csharp
+public class DbIdGraphType : ScalarGraphType
+{
+    public DbIdGraphType()
+    {
+        Name = "DbId";
+    }
 
-- `ValueConverter.ConvertTo` handles the case when `value` is an instance of the requested type by returning `value`. Therefore, `ValueConverter.ConvertTo` neatly handles both input and output representations of the scalar.
+    public override object ParseLiteral(IValue value) => value switch
+    {
+        StringValue s => int.TryParse(s.Value, out int i) && i > 0 ? i : throw new FormatException($"'{s.Value}' is not a valid identifier."),
+        null => 0,
+        _ => ThrowLiteralConversionError(value)
+    };
 
-- Since GraphQL specifies no response format, `Serialize` is not responsible for preparing the scalar for transport to the client. It is only responsible for generating an object which can eventually be serialized by `IDocumentWriter` or other transport-focused API.
+    public override object ParseValue(object value) => value switch
+    {
+        string s => int.TryParse(s, out int i) && i > 0 ? i : throw new FormatException($"'{s}' is not a valid identifier."),
+        null => 0,
+        _ => ThrowValueConversionError(value)
+    };
 
-In this example, you created a custom scalar. In summary:
+    public override object Serialize(object value) => value switch
+    {
+        int i => i > 0 ? i.ToString() : i == 0 ? null : ThrowSerializationError(value),
+        _ => ThrowSerializationError(value)
+    };
+}
+```
 
-- Create a class for the server-side representation of the scalar
-- Implement a `ScalarGraphType` which handles parsing, literal parsing, and serialization
-- Register the `ScalarGraphType with the DI container
-- Define how to parse the raw representation of the scalar to its server-side representation using `ValueConverter.Register`
-- Implement a `ValueNode<T>` class for the server-side representation
-- Implement an `IAstFromValueConverter` for wrapping the server-side representation in its `ValueNode<T>` implementation
-- Register the `IAstFromValueConverter` with the schema using `Schema.RegisterValueConverter`
+## Replacing built-in scalar types
+
+In some cases you may want or need to replace the functionality of the built-in graph types. This can
+be accomplished by running a schema node visitor on the schema after it has been initialized. Keep in
+mind that the node visitor will walk through the introspection portion of the schema, so replacing
+a built-in type may affect the operation of introspection queries.
+
+Below is a sample of how to replace the built-in `IdGraphType` with the `DbIdGraphType` shown above.
```

Comment:
The thought is correct, but I'm not sure yet it will work.

Reply:
I added a test to verify that it works.

Reply:
Sorry, I added a test to show that the `DbIdGraphType` works with nullability coercion.

Reply:
This sample was replaced with a more common scenario -- replacing the boolean graph type (which is more strict in 4.0) to a version that allows coercion from any type of value (similar to how it was supported in 3.x).

Reply:
Ok